### PR TITLE
 Separate streamly tests into another package

### PR DESCRIPTION
**Pull request by harendra-kumar**
To be able to use an internal test library, we need to have it as a
separate package. See comments in streamly-benchmarks.cabal for details.

Use a test library for common test modules
Also, added "opt" flag to control optimization levels.

Add a README for tests.
Update benchmarks readme.

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Build only
```
Comment (pranaysashank) [inline test/README.md:1]:
```suggestion
# Build only, use any of the following
```

File: benchmark/README.md [hunk 2]
```diff
@@ -1,11 +1,56 @@
-## Running Benchmarks
+## Build Only
+
+Building single benchmark suite:
```
Comment (pranaysashank) [inline benchmark/README.md:3]:
```suggestion
Building single benchmark suite: use any of the following
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -193,6 +194,7 @@ jobs:
             command: |
               apt-get install -y ghc-8.2.2
               apt-get install -y cabal-install-3.2
+              apt-get install libtinfo-dev
```
Comment (pranaysashank) [inline .circleci/config.yml:197]: what's this needed for?

**Review by pranaysashank — COMMENTED** [review]
Need to check why the coverage build is failing to run any tests.

File: .circleci/config.yml (hunk 3 above)
Comment (harendra-kumar) [reply .circleci/config.yml:197]: The linker was failing.

**Pull request merged**